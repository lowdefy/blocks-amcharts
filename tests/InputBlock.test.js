/*
  Copyright 2020 Lowdefy, Inc

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
*/

import React from 'react';
import { mockBlock, runBlockSchemaTests, runRenderTests } from '@lowdefy/block-tools';
import { configure, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
configure({ adapter: new Adapter() });

import { InputBlock } from '../src';
import examples from '../demo/examples/InputBlock.yaml';
import meta from '../src/blocks/InputBlock/InputBlock.json';

runRenderTests({ examples, Block: InputBlock, meta });
runBlockSchemaTests({ examples, meta });

const { before, methods, getProps } = mockBlock({ meta });
beforeEach(before);

it('test input change', () => {
  const block = {
    id: 'one',
    type: 'InputBlock',
  };
  let props;
  const Shell = () => {
    props = getProps(block);
    return <InputBlock {...props} methods={methods} />;
  };
  const wrapper = mount(<Shell />);
  expect(wrapper.find('[data-testid="one-input"]').prop('value')).toEqual('');
  wrapper.find('[data-testid="one-input"]').simulate('change', { target: { value: 'new value' } });
  expect(methods.setValue).toHaveBeenCalledWith('new value');
});
