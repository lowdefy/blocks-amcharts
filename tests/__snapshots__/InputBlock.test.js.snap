// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render default - value[0] 1`] = `
<label>
  Label
  :
  <input
    data-testid="default-input"
    id="default"
    onChange={[Function]}
    value=""
  />
</label>
`;

exports[`Render properties.label - value[0] 1`] = `
<label>
  Input Foo Label
  :
  <input
    data-testid="properties.label-input"
    id="properties.label"
    onChange={[Function]}
    value=""
  />
</label>
`;

exports[`Render required = true default - value[0] 1`] = `
<label>
  Label
  :
  <input
    data-testid="default-input"
    id="default"
    onChange={[Function]}
    value=""
  />
</label>
`;

exports[`Render required = true properties.label - value[0] 1`] = `
<label>
  Input Foo Label
  :
  <input
    data-testid="properties.label-input"
    id="properties.label"
    onChange={[Function]}
    value=""
  />
</label>
`;

exports[`Test Schema default 1`] = `true`;

exports[`Test Schema default 2`] = `null`;

exports[`Test Schema properties.label 1`] = `true`;

exports[`Test Schema properties.label 2`] = `null`;
